index.js

```diff
@@ -36,12 +37,31 @@ let jsonApiMiddleware = [
 
 class JsonApi {
 
-  constructor(apiUrl, middleware = jsonApiMiddleware) {
-    this._originalMiddleware = middleware.slice(0)
-    this.middleware = middleware.slice(0)
+  constructor (options = {}) {
+    if (!(arguments.length == 2 && _.isString(arguments[0]) && _.isArray(arguments[1])) && !(arguments.length === 1 && (_.isPlainObject(arguments[0]) || _.isString(arguments[0])))) {
+      throw new Error("Invalid argument, initialize Devour with an object.")
+    }
```

Comment:
Explicit argument deprecation. 

One of (String, Array) OR (String) OR (Object) as the initializer argument